### PR TITLE
[docker/ci] Build ray-ml docker image for python 3.9 and 3.10

### LINKED ISSUE #16599: [RLlib][Tune] Build docker image for Python 3.9
### What is the problem?

Currently building Ray docker image with Python 3.9 times out, because some of the direct and transitive dependencies pinned in `python/requirements/rllib/requirements_rllib.txt` and `python/requirements/tune/requirements_tune.txt` do not support Python 3.9, and the pip dependency resolver goes into an edge case.

*Ray version and other system information (Python version, TensorFlow version, OS):*

Python 3.9, Ray 1.4.1

### Reproduction (REQUIRED)
https://github.com/ray-project/ray/pull/16571/files

### PR DESCRIPTION
**Pull request by acxz**
## Related issue number

Closes #16599

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by amogkam** [comment]
Hey @acxz unfortunately this is not going to work out of the box I believe. There are some dependency issues in our ML requirements. Tracking issue here  https://github.com/ray-project/ray/issues/16599

**Comment by amogkam** [comment]
But I also don’t think building ML images should be a requirement for non-experimental support.

**Comment by acxz** [comment]
I see, I figure we can have a PR open to have the CI running against and see if the error logs have changed since then.

**Comment by acxz** [comment]
not stale

**Comment by krfricke** [comment]
Now that #28808 is merged we can give this another shot. I've just merged master, let's see how the build goes.

**Comment by krfricke** [comment]
Python 3.9 seems to pass.

For 3.10, the problem was h5py, which pinned numpy to 1.19.3, which is incompatible with Python 3.10. I've upgraded that dependency.

Additionally, the cuda images were misconfigured in the 3.10 build job previously, this is also fixed.

**Comment by krfricke** [comment]
Image builds are passing, will merge once the rest of the CI is green.

**Review by krfricke — APPROVED** [review]

**Comment by krfricke** [comment]
CI looks fine, will merge if no objections cc @amogkam

**Review by richardliaw — APPROVED** [review]

File: python/requirements/data_processing/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 dask[complete]==2021.03.0; python_version < '3.7'
-dask[complete]==2022.2.0; python_version >= '3.7'
+dask[complete]; python_version >= '3.7'
```
Comment (amogkam) [inline python/requirements/data_processing/requirements.txt:2]: lets keep this pinned to the most recent version?

File: python/requirements/ml/requirements_rllib.txt [hunk 2]
```diff
@@ -24,7 +24,7 @@ pybullet==3.2.0
 recsim==0.2.4
 tensorflow_estimator==2.9.0
 # DeepMind's OpenSpiel
-open-spiel==1.0.2
+open-spiel==1.2
```
Comment (amogkam) [inline python/requirements/ml/requirements_rllib.txt:27]: can we do an exact pin here please

Comment (amogkam) [reply python/requirements/ml/requirements_rllib.txt:27]: Ok turns out there 1.2 is the exact version for `open-spiel`

File: python/requirements/data_processing/requirements.txt (hunk 1 above)
Comment (amogkam) [inline python/requirements/data_processing/requirements.txt:2]:
```suggestion
dask[complete]==2022.10.1; python_version >= '3.7'
```

**Review by amogkam — APPROVED** [review]

File: python/requirements/data_processing/requirements.txt (hunk 1 above)
Comment (amogkam) [reply python/requirements/data_processing/requirements.txt]: updated

**Comment by acxz** [comment]
Thanks so much @krfricke and @amogkam for finishing this task up!

**Comment by krfricke** [comment]
Thanks for the initial contribution @acxz!

**Pull request merged**